turbomoleio/core/control.py

```diff
@@ -217,20 +237,57 @@ def from_string(cls, string):
             mol, fi = get_mol_and_indices_frozen("\n".join(coordinates))
             molecules.append(MoleculeSystem(molecule=mol, frozen_indices=fi))
 
-        return cls(gradients=gradients, energies=energies, molecules=molecules)
+        if lattice_string:
+            # skip the first as it is before the first
+            lattice_cycles = lattice_string.split("cycle =")[1:]
+
+            lattice_vectors = []
+            lattice_gradients = []
+            gradlatt_energies = []  # used to double check with energy from grad datagroup
+
+            for c in lattice_cycles:
+                lines = c.splitlines()
+                header = lines[0]
+                match = re.search(r"energy\s+=\s+([+-]?[0-9]*[.]?[0-9]+)", header)
+                gradlatt_energies.append(float(match.group(1)))
+
+                ndim = len(lines[1].split())
+                periodicity = f'{ndim}D'
+                lv = []
+                for l in lines[1:1+ndim]:
+                    l = l.replace("D", "E")
+                    lv.append([float(sp) for sp in l.split()])
+                lattice_vectors.append(np.array(lv))
+                lg = []
+                for l in lines[1+ndim:1+2*ndim]:
+                    l = l.replace("D", "E")
+                    lg.append([float(sp) for sp in l.split()])
+                lattice_gradients.append(np.array(lg))
+
+        return cls(gradients=gradients, energies=energies, molecules=molecules,
+                   periodicity=periodicity,
+                   lattice_vectors=lattice_vectors, lattice_gradients=lattice_gradients)
 
     @classmethod
     def from_file(cls, filename='gradient'):
         """
         Creates Gradient object reading from a given file.
 
+        Note: For periodic systems, the lattice gradients are usually either
+            in the control file directly or in a different file. One option
+            is thus to initialize the Gradients object from the control file,
+            which will then get the atom gradients from the $grad datagroup
+            and the lattice gradient from the $gradlatt datagroup.
+
         Args:
             filename (str): Name of the file from which this Gradient object
                 should be read, default is "gradient".
         """
         dg = DataGroups.from_file(filename)
         string = dg.show_data_group("grad", show_from_subfile=True, raise_if_missing_subfile=True)
-        return cls.from_string(string)
+        lattice_string = dg.show_data_group("gradlatt", show_from_subfile=True,
```

Comment:
I think we could replace the default with "control" rather safely. I would probably wait for a separate PR for this (as it "might" have implications on BASFflows even if I think it would be ok).

Reply:
Opened issue #15 .
Closing here